serving/central/src/main/java/ai/djl/serving/central/model/dto/ModelDTO.java

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package ai.djl.serving.central.model.dto;
+
+import ai.djl.repository.Artifact;
+import ai.djl.repository.License;
+import java.net.URI;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Transferring model-artifact-info.
+ *
+ * <p>this class inherits from ModelReferenceDTO and can so be used as a ModelReference.
+ *
+ * @author erik.bamberg@web.de
+ */
+public class ModelDTO extends ModelReferenceDTO {
+
+    private static final Logger logger = LoggerFactory.getLogger(ModelDTO.class);
+
+    private boolean snapshot;
+    private String description;
+    private String website;
+    private URI repositoryURI;
+    private Date lastUpdated;
+    private String resourceType;
+    private Map<String, String> licenses;
+
+    private List<FileResourceDTO> files;
+
+    private String dataset;
+    private String layers;
+    private String backbone;
+
+    private double width;
```

Comment:
@frankfliu 
do you have any idea when we will be there to have that standard argument list in the core model ? 
I think the rework on this artifact-model is important and urgent before more gets onboarded and effort gains, 

This ModelDTO was necessary cause of cyclic dependencies in Artifact/MetaData-model requires a Serializer which we had a working solution.

But apart of this.

The ModelDTO is not core-component it is just a data-transfer-object representing the view in the UI, this is the only purpose.

I would success to add a Map<String,String> additional this the properties in ModelDTO and simply hide known properties not used for the particular applicationAnd also visualise all not directly known fields.
So we can have best of both worlds.
 (having kind of CVModelDTO and NLPModelDTO would actually be nicer). 

Obviously width/height just makes sense for applications cv/* and not nlp/* , so pretty easy to hide/display them for instance. And if you have a field which won't makes sense and it is empty, well then it is as it is ;-)

Problem is to create a UI you have to visualize and group data-fields so that the user has a good user experience and find what he is looking for where he expects it => means if you  have width & height it make sense to visualize them next to each and height in the middle of a big table and width at the bottom of the table. You expect to see field X*y. 

So I would like to create a dynamic table for all unknow dynamic field and the few we know there meaning visualize them and hide/display based on the application-context.

This would at least be a start to find out and collect information about what properties we would need for a proper code-artifact-model.

Reply:
@ebamberg 
I don't expect it's going to happen any time soon. 
We should be able to use width/height etc, they are pretty much standardized.
But NLP is much more complicated, we have to keep them are name/value pair for now.

Reply:
@frankfliu that's perfectly fine with me, I will do the necessary changes.

Reply:
@frankfliu just pushed some changes, 

maybe this could be a solution for now. 
flexible name/value pairs in the property for everything that is unknown. 
and a class hierarchy for the application-type if we sort out fix properties for the application-types later on. Easy to extends.
CV-DTO still have width&height etc. to demonstration this.

and the UI picks up the flexible properties and dynamically creates field from them.